src/creature.cpp

```diff
@@ -347,11 +347,16 @@ void Creature::melee_attack(Creature &t, bool allow_special)
     melee_attack( t, allow_special, no_technique_id );
 }
 
+float Creature::hit_roll_at_accuracy( float accuracy )
```

Comment:
Please comment what this means, I suspect it really doesn't mean anything and it needs to be extracted differently.

Reply:
It's common math for all melee attacks.
It is necessary to allow different attacks to have different accuracies. `hit_roll` alone is not sufficient, multiplying the result of `hit_roll` by a value would be terrible statistics-wise.

Alternatives are:
* Optional argument for `hit_roll` for different "base accuracy". This one could be beneficial, considering that some day player will need different accuracies for different weapon selections (dual wielding, offhand kicks, headbutt when grappled and unable to use zweihander etc.)
* Inlining the roll directly, replicating the math in the instances of usage of this function
* Decoupling melee math from creature class completely, having the creature only supply accuracy and having the math for attacks be determined by some other class